manim/mobject/three_dimensions.py

```diff
@@ -189,3 +199,427 @@ def generate_points(self):
+            The radius.
+        v : :class:`float`
+            The azimuthal angle.
+        """
+        r = u
+        phi = v
+        return np.array(
+            [
+                r * np.sin(self.theta) * np.cos(phi),
+                r * np.sin(self.theta) * np.sin(phi),
+                r * np.cos(self.theta),
+            ]
+        )
+
+    def _rotate_to_direction(self):
+        x, y, z = self.direction
+
+        r = np.sqrt(x ** 2 + y ** 2 + z ** 2)
+        theta = np.arccos(z / r)
+
+        if x == 0:
+            if y == 0:  # along the z axis
+                phi = 0
+            else:
+                phi = np.arctan(np.inf)
+                if y < 0:
+                    phi += PI
+        else:
+            phi = np.arctan(y / x)
+        if x < 0:
+            phi += PI
+
+        # Undo old rotation (in reverse order)
+        self.rotate(-self._current_phi, Z_AXIS, about_point=ORIGIN)
+        self.rotate(-self._current_theta, Y_AXIS, about_point=ORIGIN)
+
+        # Do new rotation
+        self.rotate(theta, Y_AXIS, about_point=ORIGIN)
+        self.rotate(phi, Z_AXIS, about_point=ORIGIN)
+
+        # Store values
+        self._current_theta = theta
+        self._current_phi = phi
+
+    def set_direction(self, direction):
+        self.direction = direction
+        self._rotate_to_direction()
+
+    def get_direction(self):
+        return self.direction
+
+
+class Cylinder(ParametricSurface):
+    """A cylinder, defined by its height, radius and direction,
+
+    Examples
+    ---------
+    .. manim:: ExampleCylinder
+        :save_last_frame:
+
+        class ExampleCylinder(ThreeDScene):
+            def construct(self):
+                axes = ThreeDAxes()
+                cylinder = Cylinder(radius=2, height=3)
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                self.add(axes, cylinder)
+
+    Parameters
+    ---------
+    radius : :class:`float`
+        The radius of the cylinder.
+    height : :class:`float`
+        The height of the cylinder.
+   direction : :class:`numpy.array`
+       The direction of the central axis of the cylinder.
+   v_min : :class:`float`
+       The height along the height axis (given by direction) to start on.
+   v_max : :class:`float`
+       The height along the height axis (given by direction) to end on.
+   show_ends : :class:`bool`
+       Whether to show the end caps or not.
+   """
+
+   def __init__(
+       self,
+       radius=1,
+       height=2,
+       direction=Z_AXIS,
+       v_min=0,
+       v_max=TAU,
+       show_ends=True,
+       resolution=24,
+       **kwargs
+   ):
+       self._height = height
+       self.radius = radius
+       ParametricSurface.__init__(
+           self,
+           self.func,
+           resolution=resolution,
+           u_min=-self._height / 2,
+           u_max=self._height / 2,
+           v_min=v_min,
+           v_max=v_max,
+           **kwargs,
+       )
```

Comment:
This all needs a space due to another copy/paste issue I generated from my editor to github:
```suggestion
    direction : :class:`numpy.array`
        The direction of the central axis of the cylinder.
    v_min : :class:`float`
        The height along the height axis (given by direction) to start on.
    v_max : :class:`float`
        The height along the height axis (given by direction) to end on.
    show_ends : :class:`bool`
        Whether to show the end caps or not.
    """

    def __init__(
        self,
        radius=1,
        height=2,
        direction=Z_AXIS,
        v_min=0,
        v_max=TAU,
        show_ends=True,
        resolution=24,
        **kwargs
    ):
        self._height = height
        self.radius = radius
        ParametricSurface.__init__(
            self,
            self.func,
            resolution=resolution,
            u_min=-self._height / 2,
            u_max=self._height / 2,
            v_min=v_min,
            v_max=v_max,
            **kwargs,
        )
```